segmenttree/operations.go

```diff
@@ -0,0 +1,69 @@
+package segmenttree
+
+func Sum(x Addable, y Addable) Addable {
+	return x.Add(y)
+}
+
+func Count(x Addable, y Addable) Addable {
+	return x.Add(y)
+}
+
+func Average(x Addable, y Addable) Addable {
+	return x.Add(y)
+}
+
+func Min(x Comparable, y Comparable) Comparable {
+	res := x.Compare(y)
+
+	if res < 0 {
+		return x
+	}
+
+	return y
+}
+
+func Max(x Comparable, y Comparable) Comparable {
+	res := x.Compare(y)
+	if res > 0 {
+		return x
+	}
+
+	return y
+}
+
+type Comparable interface {
```

Comment:
Poor attempt to make AVG (operates on two values) treatable the same as SUM and COUNT (operate on single values)